### PR TITLE
Backport patch for 6.8.0 and add tests for future failures

### PR DESCRIPTION
**Pull request by hmaarrfk**
I thought that this would be a good test to add, it did have the desired effect, then I tried their patch and it seems ok.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -19,6 +19,13 @@ source:
...
 
+    # Fix major regression in 6.8.0
+    # https://github.com/spyder-ide/qtpy/issues/494
+    # https://github.com/conda-forge/pyside2-feedstock/issues/247
+    # https://codereview.qt-project.org/c/pyside/pyside-setup/+/597328
+    # https://bugreports.qt.io/browse/PYSIDE-2888
+    - patches/try_to_fix_star_import_crash.diff
+
 build:
   number: 0
```
Comment (hmaarrfk) [inline recipe/meta.yaml:30]:
```suggestion
  number: 1
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -148,6 +157,14 @@ test:
...
     - test ! -f $PREFIX/bin/uic  # [unix]
     - if exist %LIBRARY_BIN%\uic.exe exit 1  # [win]
+    # Star imports have broken many times on PySide6
+    # https://github.com/spyder-ide/qtpy/issues/494
+    # https://github.com/spyder-ide/qtpy/issues/480
+    # We add it here for safety so we don't release really broken packages
+    - python -c "import PySide6; from PySide6.QtCore import *"
+    # Ensure we can still import things for one of the most popular packages
+    - python -c "from qtpy import API; assert API == 'pyside6', 'PySide6 not in use'"  # [py==311]
+    - python -c "from qtpy.QtCore import Qt"                                           # [py==311]
```
Comment (ccordoba12) [inline recipe/meta.yaml:167]: Why is this test marked for 3.11 only? I think it be better to leave it for any version.

**Review by ccordoba12 — COMMENTED** [review]
Besides my tiny comment below, I have another question: why not wait for 6.8.1?

File: recipe/meta.yaml (hunk 2 above)
Comment (hmaarrfk) [reply recipe/meta.yaml:167]: I don’t like adding too many optional dependencies for these nice to have tests. 

By having it only for python 3.11 we test it, while still testing minimal dependencies in other pythons

**Comment by hmaarrfk** [comment]
>  why not wait for 6.8.1?

its hard to "merge red", i could downgrade the version, but it was "just as easy" to try the upstream patch.

**Comment by hmaarrfk** [comment]
The nice thing is that upstream seems to have added a similar test to the one I added to their automation.

**Comment by ccordoba12** [comment]
> its hard to "merge red", i could downgrade the version, but it was "just as easy" to try the upstream patch.

Ok, I was just curious.

> The nice thing is that upstream seems to have added a similar test to the one I added to their automation.

Finally! Haha

File: recipe/meta.yaml [hunk 3]
```diff
@@ -81,6 +88,8 @@ requirements:
 
 test:
   requires:
+    # Add some qtpy tests to ensure we don't fully break things
+    - qtpy                 # [py==311]
```
Comment (jschueller) [inline recipe/meta.yaml:92]: checking with qtpy is overkill imho
checking star import would suffice, but even that is already done in the pyside side:
https://github.com/pyside/pyside-setup/commit/2a2d01399ad837db17cae3c2d00446b88338a627
but its still +1 even if you want to keep it

**Review by jschueller — APPROVED** [review]
imho it was enough to check the start import, and even that is

File: recipe/meta.yaml (hunk 3 above)
Comment (hmaarrfk) [reply recipe/meta.yaml]: ok but i might add it back within the year ;)

**Comment by hmaarrfk** [comment]
For some reason the CIs didn't trigger.

I pushed directly to the `pyside6`  branch a cleanup commit to trigger things
https://github.com/conda-forge/pyside2-feedstock/commit/1e55e57a1e0744c98e02bad3a4ce3ab2114797f1

**Pull request merged**